include/amqpcpp/monitor.h

```diff
@@ -54,7 +54,7 @@ class Monitor
     Monitor(Watchable *watchable) : _watchable(watchable)
     {
         // register with the watchable
-        _watchable->add(this);
+        if (_watchable) _watchable->add(this);
```

Comment:
Did you run into an issue because of this? Afaik we never call Monitor(nullptr).

Reply:
I didn't run into actual issues. Just when looking at the code I saw that _watchable is allowed to be nullptr in all other methods/constructors and thought it is more consistent to also handle it gracefully in this constructor.

Reply:
No it is not an issue, in the constructor it can never be null (in the other methods it can because it could be set to null _after_ construction). Can you undo this?

Reply:
Ok, I removed the corresponding commit